test/basic.test.js

```diff
@@ -28,8 +28,25 @@ test('Connect-Subscribe-Publish-Disconnect 300 clients using WS and MQTT/MQTTS p
   await pMap(clients, c => c.end(), pMapOptions)
 })
 
+test('Unhautorized client', function (t) {
+  t.tearDown(helper.closeBroker)
+  t.plan(1)
+
+  const noError = t.error.bind(t)
```

Comment:
You can just use async/await.

Reply:
I have problems when using async functions with test, they end the test when they resolve and so even if I have a t.plan they doesn't wait for that to call end. In this case I have to setup a listener and without that it doesn't work

Reply:
This should work:

```js
await helper.startBroker()
var client = helper.startClient('mqtt', { username: 'user', password: 'notallowed' }, true)
var err = await once(client, 'error')
t.equal(err.code, 4, 'Connection should be rejected with code 4')
```

Reply:
You always tech me something new :fearful:

Reply:
@mcollina Where do I find `once`? `events`? becaus it isn't working